opentracing/context.go

```diff
@@ -0,0 +1,35 @@
+package opentracing
+
+// A ContextID is the smallest amount of state needed to describe a span's
+// context within a larger [potentially distributed] trace. The ContextID is
+// not intended to encode the span's operation name, timing, or log data, but
+// merely any unique identifiers (etc) needed to contextualize it within a
+// larger trace tree.
+//
+// ContextIDs are sufficient to propagate the, well, *context* of a particular
+// trace from process to process.
+type ContextID interface {
+	// Create a child context for this ContextID, and return both that child's
+	// own ContextID as well as any Tags that should be added to the child's
+	// Span.
+	//
+	// The returned ContextID type must be the same as the type of the
+	// ContextID implementation itself.
+	NewChild() (childCtx ContextID, initialChildSpanTags Tags)
+
+	// Serializes the ContextID as an arbitrary byte string.
+	Serialize() []byte
```

Comment:
> but I do think []bytes should be an option at least

Agreed, since binary and text are the two prevalent encodings. HTTP instrumentation should choose text, a thrift-based protocol might choose bytes, thus avoiding potentially unnecessary intermediate encoding.

> Some ContextIDs won't be able to represent themselves that way, so it's problematic to add it at this layer.

I was thinking as an optional interface, e.g https://github.com/uber-common/opentracing-go/blob/master/zipkin.go.

> it also makes me grumpy that Zipkin sends the parent_id over the wire

I thought that was to help with cases where some spans of the trace didn't make it to storage. Having parent span ID allows to put spans in some sort of casual relationship in the UI. It's like an error correction code. Won't help much if storage is missing two levels of the tree.